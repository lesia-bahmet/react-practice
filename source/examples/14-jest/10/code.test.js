// Core
import { v4 } from 'uuid';

global.Date = jest.fn();

class User {
    created = new Date();
    id = v4();

    constructor(name) {
        this.name = name;
    }
}

test('«User» constructor should match its snapshot inline', () => {
    expect(new User('Jack')).toMatchSnapshot();
});
