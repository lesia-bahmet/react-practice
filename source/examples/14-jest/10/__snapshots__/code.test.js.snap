// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`«User» constructor should match its snapshot inline 1`] = `
User {
  "created": mockConstructor {},
  "id": "123",
  "name": "Jack",
}
`;
